src/client/hooks/useAllowance.ts

```diff
@@ -0,0 +1,56 @@
+import { useState, useEffect, useContext } from 'react';
+import isEqual from 'lodash/isEqual';
+
+import { AppServiceContext } from '@context';
+import { NetworkSelectors, WalletSelectors } from '@store';
+import { TokenAllowance } from '@types';
+import { useAppSelector } from '@hooks';
+
+import { usePrevious } from './usePrevious';
+
+export function useAllowance(
+  tokenAddress: string | undefined,
+  vault: { address: string; token: { address: string } } | undefined
```

Comment:
We should try to use a flat object when possible and not nested objects.

Reply:
solved on fff3e86f6179d4e466b091c8f555551d66f809ef